### PR TITLE
Issue/#705 change pop time computation to include team size of queue

### LINKED ISSUE #705: Fix queue pop time calculation for tmm queues
Right now the queue pop times are calculated based on trying to maintain a specific number of parties in a matchmaker queue. What we really want to do is adjust it so it tries to maintain enough players for a specific number of possible matches. I would recommend we rename the variable `QUEUE_POP_DESIRED_PLAYERS` to `QUEUE_POP_DESIRED_MATCHES` and set it to a default of 4.

1. Fix the bug where number of parties is used instead of number of players
https://github.com/FAForever/server/blob/55c42f0a4b8795f0f927614638b93d5731916285/server/matchmaker/matchmaker_queue.py#L98
2. Include the queue's team_size in the time calculation
https://github.com/FAForever/server/blob/55c42f0a4b8795f0f927614638b93d5731916285/server/matchmaker/pop_timer.py#L74

### PR DESCRIPTION
**Pull request by Askaholic**
Change the queue pop time config to number of desired matches, and adjust the pop timer calculations accordingly. This means queues with larger team size will need more players to be in queue for the pop time to be reduced. Max pop time is still the same for all queues.

Closes #705

### CODE REVIEW

**Review by cleborys — APPROVED** [review]
Maybe that's a bit of a meta-comment, but I bet that the client showing the time remaining to the next queue pop causes the number of players queued up at a queue pop to be less linear in queue pop time than it would otherwise be, with players only joining when the timer gets somewhat low (at least I have done that in the past :blush:).

File: server/matchmaker/pop_timer.py [hunk 1]
```diff
@@ -25,8 +25,8 @@ class PopTimer(object):
     The player queue rate is based on a moving average over the last few pops.
     The exact size can be set in config.
     """
-    def __init__(self, queue_name: str):
-        self.queue_name = queue_name
+    def __init__(self, queue: "MatchmakerQueue"):
+        self.queue = queue
```
[the remark this answers is not in the document]
Comment (Askaholic) [reply server/matchmaker/pop_timer.py:29]: Yea I know

File: server/matchmaker/pop_timer.py [hunk 2]
```diff
@@ -35,16 +35,16 @@ def __init__(self, queue_name: str):
         # Optimistically schedule first pop for half of the max pop time
         self.next_queue_pop = self._last_queue_pop + (config.QUEUE_POP_TIME_MAX / 2)
 
-    async def next_pop(self, get_num_players: Callable[[], int]):
+    async def next_pop(self):
         """ Wait for the timer to pop. get_num_players needs to return the current
```
[the remark this answers is not in the document]
Comment (Askaholic) [reply server/matchmaker/pop_timer.py]: Good catch

**Review by 0x647262 — APPROVED** [review]
Code's clean, tests pass. LGTM!

File: server/matchmaker/pop_timer.py [hunk 3]
```diff
@@ -66,17 +65,18 @@ def time_until_next_pop(self, num_queued: int, time_queued: float) -> float:
...
         if total_times:
             self._logger.debug(
-                "Queue rate for %s: %f/s", self.queue_name,
+                "Queue rate for %s: %f/s", self.queue.name,
                 total_players / total_times
             )
 
         # Obtained by solving $ NUM_PLAYERS = rate * time $ for time.
-        next_pop_time = config.QUEUE_POP_DESIRED_PLAYERS * total_times / total_players
+        desired_players = config.QUEUE_POP_DESIRED_MATCHES * self.queue.team_size
```
Comment (Askaholic) [inline server/matchmaker/pop_timer.py:73]: Oops, number of players per match is team_size x2
```suggestion
        desired_players = config.QUEUE_POP_DESIRED_MATCHES * self.queue.team_size * 2
```

**Comment by Sheikah45** [comment]
I suspect this may have had an unintended effect on the queue pop times as after the server update I am receiving pop times that are in the past. However this is only occuring on the 2v2 matchmaker. 1v1 is fine.

**Pull request merged**